### ISSUE TITLE
keadm init fail because use apiextensions.k8s.io/v1 CustomResourceDefinition

### ISSUE THREAD

**Issue by Justin-Young** [kubeedge/kubeedge#2795]
**What happened and what you expected to happen**:
Hello,
After reset cloudcore, I wanna init again, but because something with api v1 CustomReSourceDefinition is wrong, init failed. I wanna know how to solve this problem, thank you!
![image](https://user-images.githubusercontent.com/53030626/115537255-38362500-a2cd-11eb-82d9-4fd0f4304513.png)
```
Kubernetes version verification passed, KubeEdge installation will start...
W0421 16:42:50.741365   13059 warnings.go:67] apiextensions.k8s.io/v1beta1 CustomResourceDefinition is deprecated in v1.16+, unavailable in v1.22+; use apiextensions.k8s.io/v1 CustomResourceDefinition
Error: CustomResourceDefinition.apiextensions.k8s.io "devices.devices.kubeedge.io" is invalid: spec.versions: Invalid value: []apiextensions.CustomResourceDefinitionVersion{apiextensions.CustomResourceDefinitionVersion{Name:"v1alpha2", Served:true, Storage:true, Deprecated:false, DeprecationWarning:(*string)(nil), Schema:(*apiextensions.CustomResourceValidation)(0xc011840dd0), Subresources:(*apiextensions.CustomResourceSubresources)(nil), AdditionalPrinterColumns:[]apiextensions.CustomResourceColumnDefinition(nil)}}: per-version schemas may not all be set to identical values (top-level validation should be used instead)
```
By the way, I have another problem, because of which I wanna reset cloudcore, that is the edgecore can‘t join with cloudcore, and the error message is `"Error: failed to get edge certificate from the cloudcore, error: Get "https://192.168.56.102:10002/edge.crt": x509: certificate is valid for 10.0.2.15, not 192.168.56.102"`. Can I use the flag advertise-address to solve this problem?
Part of the log information is as follows：
```
Apr 21 15:59:55 node1 systemd[1]: edgecore.service holdoff time over, scheduling restart.
Apr 21 15:59:55 node1 systemd[1]: Stopped edgecore.service.
Apr 21 15:59:55 node1 systemd[1]: Started edgecore.service.
Apr 21 15:59:55 node1 edgecore[4520]: INFO: Install client plugin, protocol: rest
Apr 21 15:59:55 node1 edgecore[4520]: INFO: Installed service discovery plugin: edge
Apr 21 15:59:59 node1 edgecore[4520]: I0421 15:59:59.036008    4520 server.go:72] Version: v1.6.1
Apr 21 15:59:59 node1 edgecore[4520]: I0421 15:59:59.052785    4520 sql.go:21] Begin to register twin db model
Apr 21 15:59:59 node1 edgecore[4520]: I0421 15:59:59.053024    4520 module.go:34] Module twin registered successfully
Apr 21 15:59:59 node1 edgecore[4520]: I0421 15:59:59.071836    4520 client.go:77] Connecting to docker on unix:///var/run/docker.sock
Apr 21 15:59:59 node1 edgecore[4520]: I0421 15:59:59.071940    4520 client.go:94] Start docker client with request timeout=0s
Apr 21 15:59:59 node1 edgecore[4520]: I0421 15:59:59.102445    4520 docker_service.go:241] Hairpin mode set to "hairpin-veth"
Apr 21 15:59:59 node1 edgecore[4520]: I0421 15:59:59.157193    4520 docker_service.go:256] Docker cri networking managed by kubernetes.io/no-op
Apr 21 15:59:59 node1 edgecore[4520]: I0421 15:59:59.186137    4520 docker_service.go:261] Docker Info: &{ID:JLNR:33EP:JRL5:BHGG:RKG5:QBTK:WIFP:LQH7:OV6D:6IR3:HGK6:22SE Containers:1 ContainersRunning:0 ContainersPaused:0 ContainersStopped:1 Images:6 Driver:overlay2 DriverStatus:[[Backing Filesystem xfs] [Supports d_type true] [Native Overlay Diff true]] SystemStatus:[] Plugins:{Volume:[local] Network:[bridge host ipvlan macvlan null overlay] Authorization:[] Log:[awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog]} MemoryLimit:true SwapLimit:true KernelMemory:true KernelMemoryTCP:true CPUCfsPeriod:true CPUCfsQuota:true CPUShares:true CPUSet:true PidsLimit:true IPv4Forwarding:true BridgeNfIptables:true BridgeNfIP6tables:true Debug:false NFd:21 OomKillDisable:true NGoroutines:34 SystemTime:2021-04-21T15:59:59.159239763+08:00 LoggingDriver:json-file CgroupDriver:cgroupfs NEventsListener:0 KernelVersion:3.10.0-1160.el7.x86_64 OperatingSystem:CentOS Linux 7 (Core) OSType:linux Architecture:x86_64 IndexServerAddress:https://index.docker.io/v1/ RegistryConfig:0xc000557810 NCPU:2 MemTotal:1927094272 GenericResources:[] DockerRootDir:/var/lib/docker HTTPProxy: HTTPSProxy: NoProxy: Name:node1 Labels:[] ExperimentalBuild:false ServerVersion:19.03.0 ClusterStore: ClusterAdvertise: Runtimes:map[runc:{Path:runc Args:[]}] DefaultRuntime:runc Swarm:{NodeID: NodeAddr: LocalNodeState:inactive ControlAvailable:false Error: RemoteManagers:[] Nodes:0 Managers:0 Cluster:<nil> Warnings:[]} LiveRestoreEnabled:false Isolation: InitBinary:docker-init ContainerdCommit:{ID:05f951a3781f4f2c1911b05e61c160e9c30eaa8e Expected:05f951a3781f4f2c1911b05e61c160e9c30eaa8e} RuncCommit:{ID:12644e614e25b05da6fd08a38ffa0cfe1903fdec Expected:12644e614e25b05da6fd08a38ffa0cfe1903fdec} InitCommit:{ID:fec3683 Expected:fec3683} SecurityOptions:[name=seccomp,profile=default] ProductLicense: Warnings:[]}
Apr 21 15:59:59 node1 edgecore[4520]: I0421 15:59:59.186268    4520 docker_service.go:274] Setting cgroupDriver to cgroupfs
Apr 21 15:59:59 node1 edgecore[4520]: I0421 15:59:59.186362    4520 edged.go:522] RemoteRuntimeEndpoint: "unix:///var/run/dockershim.sock", remoteImageEndpoint: "unix:///var/run/dockershim.sock"
Apr 21 15:59:59 node1 edgecore[4520]: I0421 15:59:59.186392    4520 edged.go:525] Starting the GRPC server for the docker CRI shim.
Apr 21 15:59:59 node1 edgecore[4520]: I0421 15:59:59.223848    4520 remote_runtime.go:59] parsed scheme: ""
Apr 21 15:59:59 node1 edgecore[4520]: I0421 15:59:59.223880    4520 remote_runtime.go:59] scheme "" not registered, fallback to default scheme
Apr 21 15:59:59 node1 edgecore[4520]: I0421 15:59:59.223937    4520 passthrough.go:48] ccResolverWrapper: sending update to cc: {[{/var/run/dockershim.sock  <nil> 0 <nil>}] <nil> <nil>}
Apr 21 15:59:59 node1 edgecore[4520]: I0421 15:59:59.223960    4520 clientconn.go:948] ClientConn switching balancer to "pick_first"
Apr 21 15:59:59 node1 edgecore[4520]: I0421 15:59:59.224029    4520 remote_image.go:50] parsed scheme: ""
Apr 21 15:59:59 node1 edgecore[4520]: I0421 15:59:59.224040    4520 remote_image.go:50] scheme "" not registered, fallback to default scheme
Apr 21 15:59:59 node1 edgecore[4520]: I0421 15:59:59.224053    4520 passthrough.go:48] ccResolverWrapper: sending update to cc: {[{/var/run/dockershim.sock  <nil> 0 <nil>}] <nil> <nil>}
Apr 21 15:59:59 node1 edgecore[4520]: I0421 15:59:59.224063    4520 clientconn.go:948] ClientConn switching balancer to "pick_first"
Apr 21 15:59:59 node1 edgecore[4520]: W0421 15:59:59.352455    4520 nvidia.go:61] NVIDIA GPU metrics will not be available: no NVIDIA devices found
Apr 21 15:59:59 node1 edgecore[4520]: I0421 15:59:59.419110    4520 kuberuntime_manager.go:214] Container runtime docker initialized, version: 19.03.0, apiVersion: 1.40.0
Apr 21 15:59:59 node1 edgecore[4520]: I0421 15:59:59.419161    4520 edged.go:624] --cgroups-per-qos enabled, but --cgroup-root was not specified.  defaulting to /
Apr 21 15:59:59 node1 edgecore[4520]: I0421 15:59:59.436348    4520 container_manager_linux.go:276] container manager verified user specified cgroup-root exists: []
Apr 21 15:59:59 node1 edgecore[4520]: I0421 15:59:59.436395    4520 container_manager_linux.go:281] Creating Container Manager object based on Node Config: {RuntimeCgroupsName: SystemCgroupsName: KubeletCgroupsName: ContainerRuntime:docker CgroupsPerQOS:true CgroupRoot:/ CgroupDriver:cgroupfs KubeletRootDir:/var/lib/edged ProtectKernelDefaults:false NodeAllocatableConfig:{KubeReservedCgroupName: SystemReservedCgroupName: ReservedSystemCPUs: EnforceNodeAllocatable:map[] KubeReserved:map[] SystemReserved:map[] HardEvictionThresholds:[]} QOSReserved:map[] ExperimentalCPUManagerPolicy:none ExperimentalCPUManagerReconcilePeriod:0s ExperimentalPodPidsLimit:0 EnforceCPULimits:false CPUCFSQuotaPeriod:0s ExperimentalTopologyManagerPolicy:none}
Apr 21 15:59:59 node1 edgecore[4520]: I0421 15:59:59.436465    4520 topology_manager.go:126] [topologymanager] Creating topology manager with none policy
Apr 21 15:59:59 node1 edgecore[4520]: I0421 15:59:59.436482    4520 container_manager_linux.go:311] [topologymanager] Initializing Topology Manager with none policy
Apr 21 15:59:59 node1 edgecore[4520]: I0421 15:59:59.436491    4520 container_manager_linux.go:316] Creating device plugin manager: false
Apr 21 15:59:59 node1 edgecore[4520]: I0421 15:59:59.436660    4520 module.go:34] Module edged registered successfully
Apr 21 15:59:59 node1 edgecore[4520]: I0421 15:59:59.436684    4520 module.go:34] Module websocket registered successfully
Apr 21 15:59:59 node1 edgecore[4520]: I0421 15:59:59.436697    4520 module.go:34] Module eventbus registered successfully
Apr 21 15:59:59 node1 edgecore[4520]: I0421 15:59:59.437128    4520 module.go:34] Module edgemesh registered successfully
Apr 21 15:59:59 node1 edgecore[4520]: I0421 15:59:59.437155    4520 metamanager.go:45] Begin to register metaManager db model
Apr 21 15:59:59 node1 edgecore[4520]: I0421 15:59:59.437270    4520 module.go:34] Module metaManager registered successfully
Apr 21 15:59:59 node1 edgecore[4520]: W0421 15:59:59.437297    4520 module.go:37] Module servicebus is disabled, do not register
Apr 21 15:59:59 node1 edgecore[4520]: W0421 15:59:59.437318    4520 module.go:37] Module edgestream is disabled, do not register
Apr 21 15:59:59 node1 edgecore[4520]: W0421 15:59:59.437333    4520 module.go:37] Module testManager is disabled, do not register
Apr 21 15:59:59 node1 edgecore[4520]: table `device` already exists, skip
Apr 21 15:59:59 node1 edgecore[4520]: table `device_attr` already exists, skip
Apr 21 15:59:59 node1 edgecore[4520]: table `device_twin` already exists, skip
Apr 21 15:59:59 node1 edgecore[4520]: table `sub_topics` already exists, skip
Apr 21 15:59:59 node1 edgecore[4520]: table `meta` already exists, skip
Apr 21 15:59:59 node1 edgecore[4520]: table `meta_v2` already exists, skip
Apr 21 15:59:59 node1 edgecore[4520]: I0421 15:59:59.439762    4520 core.go:24] Starting module edged
Apr 21 15:59:59 node1 edgecore[4520]: I0421 15:59:59.439817    4520 core.go:24] Starting module websocket
Apr 21 15:59:59 node1 edgecore[4520]: I0421 15:59:59.439894    4520 core.go:24] Starting module eventbus
Apr 21 15:59:59 node1 edgecore[4520]: I0421 15:59:59.439943    4520 core.go:24] Starting module edgemesh
Apr 21 15:59:59 node1 edgecore[4520]: I0421 15:59:59.439986    4520 core.go:24] Starting module metaManager
Apr 21 15:59:59 node1 edgecore[4520]: I0421 15:59:59.440064    4520 core.go:24] Starting module twin
Apr 21 15:59:59 node1 edgecore[4520]: I0421 15:59:59.440147    4520 edged.go:289] Starting edged...
Apr 21 15:59:59 node1 edgecore[4520]: E0421 15:59:59.440239    4520 csi_plugin.go:226] kubernetes.io/csi: CSIDriverLister not found on KubeletVolumeHost
Apr 21 15:59:59 node1 edgecore[4520]: I0421 15:59:59.440293    4520 fs_resource_analyzer.go:64] Starting FS ResourceAnalyzer
Apr 21 15:59:59 node1 edgecore[4520]: I0421 15:59:59.440589    4520 http.go:40] tlsConfig InsecureSkipVerify true
Apr 21 15:59:59 node1 edgecore[4520]: I0421 15:59:59.441498    4520 common.go:96] start connect to mqtt server with client id: hub-client-sub-1618991999
Apr 21 15:59:59 node1 edgecore[4520]: I0421 15:59:59.441535    4520 common.go:98] client hub-client-sub-1618991999 isconnected: false
Apr 21 15:59:59 node1 edgecore[4520]: I0421 15:59:59.442205    4520 log.go:181] DEBUG: Installed strategy plugin: [RoundRobin].
Apr 21 15:59:59 node1 edgecore[4520]: I0421 15:59:59.442256    4520 log.go:181] DEBUG: ConfigurationFactory Initiated
Apr 21 15:59:59 node1 edgecore[4520]: I0421 15:59:59.442279    4520 log.go:181] INFO: Configuration files: []
Apr 21 15:59:59 node1 edgecore[4520]: I0421 15:59:59.442317    4520 log.go:181] WARN: empty configurtion from [FileSource]
Apr 21 15:59:59 node1 edgecore[4520]: I0421 15:59:59.442332    4520 log.go:181] INFO: invoke dynamic handler:FileSource
Apr 21 15:59:59 node1 edgecore[4520]: I0421 15:59:59.442376    4520 log.go:181] INFO: archaius init success
Apr 21 15:59:59 node1 edgecore[4520]: I0421 15:59:59.442664    4520 process.go:113] Begin to sync sqlite
Apr 21 15:59:59 node1 edgecore[4520]: I0421 15:59:59.443513    4520 log.go:181] INFO: create new watcher
Apr 21 15:59:59 node1 edgecore[4520]: I0421 15:59:59.445031    4520 client.go:150] finish hub-client sub
Apr 21 15:59:59 node1 edgecore[4520]: I0421 15:59:59.445079    4520 common.go:96] start connect to mqtt server with client id: hub-client-pub-1618991999
Apr 21 15:59:59 node1 edgecore[4520]: I0421 15:59:59.445096    4520 common.go:98] client hub-client-pub-1618991999 isconnected: false
Apr 21 15:59:59 node1 edgecore[4520]: I0421 15:59:59.446427    4520 client.go:166] finish hub-client pub
Apr 21 15:59:59 node1 edgecore[4520]: I0421 15:59:59.446447    4520 eventbus.go:63] Init Sub And Pub Client for externel mqtt broker tcp://127.0.0.1:1883 successfully
Apr 21 15:59:59 node1 edgecore[4520]: I0421 15:59:59.446836    4520 client.go:91] edge-hub-cli subscribe topic to $hw/events/upload/#
Apr 21 15:59:59 node1 edgecore[4520]: I0421 15:59:59.466517    4520 client.go:91] edge-hub-cli subscribe topic to $hw/events/device/+/state/update
Apr 21 15:59:59 node1 edgecore[4520]: I0421 15:59:59.490387    4520 client.go:91] edge-hub-cli subscribe topic to $hw/events/device/+/twin/+
Apr 21 15:59:59 node1 edgecore[4520]: I0421 15:59:59.494402    4520 proxy.go:95] [EdgeMesh] chain EDGE-MESH not exists
Apr 21 15:59:59 node1 edgecore[4520]: I0421 15:59:59.503015    4520 proxy.go:103] [EdgeMesh] inbound rule -p tcp -d 9.251.0.0/16 -i docker0 -j EDGE-MESH not exists
Apr 21 15:59:59 node1 edgecore[4520]: I0421 15:59:59.505491    4520 client.go:91] edge-hub-cli subscribe topic to $hw/events/node/+/membership/get
Apr 21 15:59:59 node1 edgecore[4520]: I0421 15:59:59.513990    4520 client.go:91] edge-hub-cli subscribe topic to SYS/dis/upload_records
Apr 21 15:59:59 node1 edgecore[4520]: I0421 15:59:59.520181    4520 client.go:91] edge-hub-cli subscribe topic to +/user/#
Apr 21 15:59:59 node1 edgecore[4520]: I0421 15:59:59.521755    4520 client.go:99] list edge-hub-cli-topics status, no record, skip sync
Apr 21 15:59:59 node1 edgecore[4520]: F0421 15:59:59.530441    4520 certmanager.go:91] Error: failed to get edge certificate from the cloudcore, error: Get "https://192.168.56.102:10002/edge.crt": x509: certificate is valid for 10.0.2.15, not 192.168.56.102
Apr 21 15:59:59 node1 edgecore[4520]: goroutine 122 [running]:
Apr 21 15:59:59 node1 edgecore[4520]: k8s.io/klog/v2.stacks(0xc000122001, 0xc000a0b200, 0xdc, 0x222)
Apr 21 15:59:59 node1 edgecore[4520]: /root/codes/src/github.com/kubeedge/kubeedge/vendor/k8s.io/klog/v2/klog.go:996 +0xb9
Apr 21 15:59:59 node1 edgecore[4520]: k8s.io/klog/v2.(*loggingT).output(0x4f88ec0, 0xc000000003, 0x0, 0x0, 0xc00047e850, 0x4c39a84, 0xe, 0x5b, 0x0)
Apr 21 15:59:59 node1 edgecore[4520]: /root/codes/src/github.com/kubeedge/kubeedge/vendor/k8s.io/klog/v2/klog.go:945 +0x191
Apr 21 15:59:59 node1 edgecore[4520]: k8s.io/klog/v2.(*loggingT).printf(0x4f88ec0, 0x3, 0x0, 0x0, 0x3785f60, 0x9, 0xc000a1bdc0, 0x1, 0x1)
Apr 21 15:59:59 node1 edgecore[4520]: /root/codes/src/github.com/kubeedge/kubeedge/vendor/k8s.io/klog/v2/klog.go:733 +0x17a
Apr 21 15:59:59 node1 edgecore[4520]: k8s.io/klog/v2.Fatalf(...)
Apr 21 15:59:59 node1 edgecore[4520]: /root/codes/src/github.com/kubeedge/kubeedge/vendor/k8s.io/klog/v2/klog.go:1456
Apr 21 15:59:59 node1 edgecore[4520]: github.com/kubeedge/kubeedge/edge/pkg/edgehub/certificate.(*CertManager).Start(0xc0006bbc70)
Apr 21 15:59:59 node1 edgecore[4520]: /root/codes/src/github.com/kubeedge/kubeedge/edge/pkg/edgehub/certificate/certmanager.go:91 +0xed
Apr 21 15:59:59 node1 edgecore[4520]: github.com/kubeedge/kubeedge/edge/pkg/edgehub.(*EdgeHub).Start(0xc0006bbc70)
Apr 21 15:59:59 node1 edgecore[4520]: /root/codes/src/github.com/kubeedge/kubeedge/edge/pkg/edgehub/edgehub.go:65 +0x425
Apr 21 15:59:59 node1 edgecore[4520]: created by github.com/kubeedge/beehive/pkg/core.StartModules
Apr 21 15:59:59 node1 edgecore[4520]: /root/codes/src/github.com/kubeedge/kubeedge/vendor/github.com/kubeedge/beehive/pkg/core/core.go:23 +0x15c
Apr 21 15:59:59 node1 edgecore[4520]: goroutine 1 [chan receive]:
Apr 21 15:59:59 node1 edgecore[4520]: github.com/kubeedge/beehive/pkg/core.GracefulShutdown()
Apr 21 15:59:59 node1 edgecore[4520]: /root/codes/src/github.com/kubeedge/kubeedge/vendor/github.com/kubeedge/beehive/pkg/core/core.go:34 +0x17f
Apr 21 15:59:59 node1 edgecore[4520]: github.com/kubeedge/beehive/pkg/core.Run()
Apr 21 15:59:59 node1 edgecore[4520]: /root/codes/src/github.com/kubeedge/kubeedge/vendor/github.com/kubeedge/beehive/pkg/core/core.go:51 +0x2a
Apr 21 15:59:59 node1 edgecore[4520]: github.com/kubeedge/kubeedge/edge/cmd/edgecore/app.NewEdgeCoreCommand.func1(0xc0006f4dc0, 0x4fbc000, 0x0, 0x0)
Apr 21 15:59:59 node1 edgecore[4520]: /root/codes/src/github.com/kubeedge/kubeedge/edge/cmd/edgecore/app/server.go:95 +0x3c5
Apr 21 15:59:59 node1 edgecore[4520]: github.com/spf13/cobra.(*Command).execute(0xc0006f4dc0, 0xc000112190, 0x0, 0x0, 0xc0006f4dc0, 0xc000112190)
Apr 21 15:59:59 node1 edgecore[4520]: /root/codes/src/github.com/kubeedge/kubeedge/vendor/github.com/spf13/cobra/command.go:846 +0x2c2
Apr 21 15:59:59 node1 edgecore[4520]: github.com/spf13/cobra.(*Command).ExecuteC(0xc0006f4dc0, 0xc000102058, 0x0, 0x0)
Apr 21 15:59:59 node1 edgecore[4520]: /root/codes/src/github.com/kubeedge/kubeedge/vendor/github.com/spf13/cobra/command.go:950 +0x375
Apr 21 15:59:59 node1 edgecore[4520]: github.com/spf13/cobra.(*Command).Execute(...)
Apr 21 15:59:59 node1 edgecore[4520]: /root/codes/src/github.com/kubeedge/kubeedge/vendor/github.com/spf13/cobra/command.go:887
Apr 21 15:59:59 node1 edgecore[4520]: main.main()
Apr 21 15:59:59 node1 edgecore[4520]: /root/codes/src/github.com/kubeedge/kubeedge/edge/cmd/edgecore/edgecore.go:16 +0x65
Apr 21 15:59:59 node1 edgecore[4520]: goroutine 19 [chan receive]:
Apr 21 15:59:59 node1 edgecore[4520]: k8s.io/klog/v2.(*loggingT).flushDaemon(0x4f88ec0)
Apr 21 15:59:59 node1 edgecore[4520]: /root/codes/src/github.com/kubeedge/kubeedge/vendor/k8s.io/klog/v2/klog.go:1131 +0x8b
Apr 21 15:59:59 node1 edgecore[4520]: created by k8s.io/klog/v2.init.0
Apr 21 15:59:59 node1 edgecore[4520]: /root/codes/src/github.com/kubeedge/kubeedge/vendor/k8s.io/klog/v2/klog.go:416 +0xd8
Apr 21 15:59:59 node1 edgecore[4520]: goroutine 61 [chan receive]:
Apr 21 15:59:59 node1 edgecore[4520]: k8s.io/client-go/util/workqueue.(*Type).updateUnfinishedWorkLoop(0xc0003dc900)
Apr 21 15:59:59 node1 edgecore[4520]: /root/codes/src/github.com/kubeedge/kubeedge/vendor/k8s.io/client-go/util/workqueue/queue.go:198 +0xac
Apr 21 15:59:59 node1 edgecore[4520]: created by k8s.io/client-go/util/workqueue.newQueue
Apr 21 15:59:59 node1 edgecore[4520]: /root/codes/src/github.com/kubeedge/kubeedge/vendor/k8s.io/client-go/util/workqueue/queue.go:58 +0x135
Apr 21 15:59:59 node1 edgecore[4520]: goroutine 60 [chan receive]:
Apr 21 15:59:59 node1 edgecore[4520]: k8s.io/client-go/util/workqueue.(*Type).updateUnfinishedWorkLoop(0xc0003dc8a0)
Apr 21 15:59:59 node1 edgecore[4520]: /root/codes/src/github.com/kubeedge/kubeedge/vendor/k8s.io/client-go/util/workqueue/queue.go:198 +0xac
Apr 21 15:59:59 node1 edgecore[4520]: created by k8s.io/client-go/util/workqueue.newQueue
Apr 21 15:59:59 node1 edgecore[4520]: /root/codes/src/github.com/kubeedge/kubeedge/vendor/k8s.io/client-go/util/workqueue/queue.go:58 +0x135
Apr 21 15:59:59 node1 edgecore[4520]: goroutine 49 [select]:
Apr 21 15:59:59 node1 edgecore[4520]: github.com/go-chassis/go-chassis/core/registry.(*HealthChecker).wait(0x4f79650)
Apr 21 15:59:59 node1 edgecore[4520]: /root/codes/src/github.com/kubeedge/kubeedge/vendor/github.com/go-chassis/go-chassis/core/registry/healthz.go:79 +0xf6
Apr 21 15:59:59 node1 edgecore[4520]: created by github.com/go-chassis/go-chassis/core/registry.(*HealthChecker).Run
Apr 21 15:59:59 node1 systemd[1]: edgecore.service: main process exited, code=exited, status=255/n/a
Apr 21 15:59:59 node1 edgecore[4520]: /root/codes/src/github.com/kubeedge/kubeedge/vendor/github.com/go-chassis/go-chassis/core/registry/healthz.go:62 +0xa5
Apr 21 15:59:59 node1 edgecore[4520]: goroutine 66 [chan receive]:
Apr 21 15:59:59 node1 edgecore[4520]: github.com/go-chassis/go-chassis/core/registry.(*HealthChecker).check(0x4f79650)
Apr 21 15:59:59 node1 edgecore[4520]: /root/codes/src/github.com/kubeedge/kubeedge/vendor/github.com/go-chassis/go-chassis/core/registry/healthz.go:96 +0x685
Apr 21 15:59:59 node1 edgecore[4520]: created by github.com/go-chassis/go-chassis/core/registry.(*HealthChecker).Run
Apr 21 15:59:59 node1 edgecore[4520]: /root/codes/src/github.com/kubeedge/kubeedge/vendor/github.com/go-chassis/go-chassis/core/registry/healthz.go:63 +0xc7
Apr 21 15:59:59 node1 edgecore[4520]: goroutine 67 [select]:
Apr 21 15:59:59 node1 edgecore[4520]: github.com/patrickmn/go-cache.(*janitor).Run(0xc0007b7700, 0xc000402cc0)
Apr 21 15:59:59 node1 edgecore[4520]: /root/codes/src/github.com/kubeedge/kubeedge/vendor/github.com/patrickmn/go-cache/cache.go:1079 +0xda
Apr 21 15:59:59 node1 edgecore[4520]: created by github.com/patrickmn/go-cache.runJanitor
Apr 21 15:59:59 node1 edgecore[4520]: /root/codes/src/github.com/kubeedge/kubeedge/vendor/github.com/patrickmn/go-cache/cache.go:1099 +0xa9
Apr 21 15:59:59 node1 edgecore[4520]: goroutine 68 [select]:
Apr 21 15:59:59 node1 edgecore[4520]: github.com/patrickmn/go-cache.(*janitor).Run(0xc0007b7710, 0xc000402dc0)
Apr 21 15:59:59 node1 edgecore[4520]: /root/codes/src/github.com/kubeedge/kubeedge/vendor/github.com/patrickmn/go-cache/cache.go:1079 +0xda
Apr 21 15:59:59 node1 edgecore[4520]: created by github.com/patrickmn/go-cache.runJanitor
Apr 21 15:59:59 node1 edgecore[4520]: /root/codes/src/github.com/kubeedge/kubeedge/vendor/github.com/patrickmn/go-cache/cache.go:1099 +0xa9
Apr 21 15:59:59 node1 edgecore[4520]: goroutine 69 [select]:
Apr 21 15:59:59 node1 edgecore[4520]: k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0x38b1348, 0x3b25900, 0xc000a0c8d0, 0x1, 0xc0001020c0)
Apr 21 15:59:59 node1 edgecore[4520]: /root/codes/src/github.com/kubeedge/kubeedge/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:167 +0x149
Apr 21 15:59:59 node1 edgecore[4520]: k8s.io/apimachinery/pkg/util/wait.JitterUntil(0x38b1348, 0x12a05f200, 0x0, 0x1, 0xc0001020c0)
Apr 21 15:59:59 node1 edgecore[4520]: /root/codes/src/github.com/kubeedge/kubeedge/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133 +0x98
Apr 21 15:59:59 node1 edgecore[4520]: k8s.io/apimachinery/pkg/util/wait.Until(...)
Apr 21 15:59:59 node1 edgecore[4520]: /root/codes/src/github.com/kubeedge/kubeedge/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:90
Apr 21 15:59:59 node1 edgecore[4520]: k8s.io/apimachinery/pkg/util/wait.Forever(0x38b1348, 0x12a05f200)
Apr 21 15:59:59 node1 edgecore[4520]: /root/codes/src/github.com/kubeedge/kubeedge/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:81 +0x4f
Apr 21 15:59:59 node1 systemd[1]: Unit edgecore.service entered failed state.
Apr 21 15:59:59 node1 edgecore[4520]: created by k8s.io/component-base/logs.InitLogs
Apr 21 15:59:59 node1 edgecore[4520]: /root/codes/src/github.com/kubeedge/kubeedge/vendor/k8s.io/component-base/logs/logs.go:58 +0x8a
Apr 21 15:59:59 node1 edgecore[4520]: goroutine 94 [IO wait]:
Apr 21 15:59:59 node1 edgecore[4520]: internal/poll.runtime_pollWait(0x7f6c1c514ff8, 0x72, 0x0)
Apr 21 15:59:59 node1 edgecore[4520]: /root/.gvm/gos/go1.15/src/runtime/netpoll.go:220 +0x55
Apr 21 15:59:59 node1 edgecore[4520]: internal/poll.(*pollDesc).wait(0xc0000b8b18, 0x72, 0x0, 0x0, 0x37828f1)
Apr 21 15:59:59 node1 edgecore[4520]: /root/.gvm/gos/go1.15/src/internal/poll/fd_poll_runtime.go:87 +0x45
Apr 21 15:59:59 node1 edgecore[4520]: internal/poll.(*pollDesc).waitRead(...)
Apr 21 15:59:59 node1 edgecore[4520]: /root/.gvm/gos/go1.15/src/internal/poll/fd_poll_runtime.go:92
Apr 21 15:59:59 node1 edgecore[4520]: internal/poll.(*FD).Accept(0xc0000b8b00, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
Apr 21 15:59:59 node1 edgecore[4520]: /root/.gvm/gos/go1.15/src/internal/poll/fd_unix.go:394 +0x1fc
Apr 21 15:59:59 node1 edgecore[4520]: net.(*netFD).accept(0xc0000b8b00, 0xc000075d78, 0xc000066800, 0x7f6c1c6777d0)
Apr 21 15:59:59 node1 edgecore[4520]: /root/.gvm/gos/go1.15/src/net/fd_unix.go:172 +0x45
Apr 21 15:59:59 node1 edgecore[4520]: net.(*TCPListener).accept(0xc00096b040, 0xc000075db8, 0x11d2dd8, 0x30)
Apr 21 15:59:59 node1 edgecore[4520]: /root/.gvm/gos/go1.15/src/net/tcpsock_posix.go:139 +0x32
Apr 21 15:59:59 node1 edgecore[4520]: net.(*TCPListener).Accept(0xc00096b040, 0x35b1e20, 0xc0008793e0, 0x337bea0, 0x4ea6030)
Apr 21 15:59:59 node1 edgecore[4520]: /root/.gvm/gos/go1.15/src/net/tcpsock.go:261 +0x65
Apr 21 15:59:59 node1 edgecore[4520]: net/http.(*Server).Serve(0xc0001f82a0, 0x3ba1fc0, 0xc00096b040, 0x0, 0x0)
Apr 21 15:59:59 node1 edgecore[4520]: /root/.gvm/gos/go1.15/src/net/http/server.go:2937 +0x266
Apr 21 15:59:59 node1 edgecore[4520]: k8s.io/kubernetes/pkg/kubelet/cri/streaming.(*server).Start(0xc000aa2900, 0x1, 0x275111e, 0x1205f76)
Apr 21 15:59:59 node1 edgecore[4520]: /root/codes/src/github.com/kubeedge/kubeedge/vendor/k8s.io/kubernetes/pkg/kubelet/cri/streaming/server.go:248 +0x14a
Apr 21 15:59:59 node1 edgecore[4520]: k8s.io/kubernetes/pkg/kubelet/dockershim.(*dockerService).Start.func1(0xc0006d23c0)
Apr 21 15:59:59 node1 edgecore[4520]: /root/codes/src/github.com/kubeedge/kubeedge/vendor/k8s.io/kubernetes/pkg/kubelet/dockershim/docker_service.go:415 +0x3d
Apr 21 15:59:59 node1 edgecore[4520]: created by k8s.io/kubernetes/pkg/kubelet/dockershim.(*dockerService).Start
Apr 21 15:59:59 node1 edgecore[4520]: /root/codes/src/github.com/kubeedge/kubeedge/vendor/k8s.io/kubernetes/pkg/kubelet/dockershim/docker_service.go:414 +0x86
Apr 21 15:59:59 node1 edgecore[4520]: goroutine 106 [select]:
Apr 21 15:59:59 node1 edgecore[4520]: google.golang.org/grpc.(*ccBalancerWrapper).watcher(0xc000402b80)
Apr 21 15:59:59 node1 edgecore[4520]: /root/codes/src/github.com/kubeedge/kubeedge/vendor/google.golang.org/grpc/balancer_conn_wrappers.go:69 +0xc8
Apr 21 15:59:59 node1 systemd[1]: edgecore.service failed.
```
..
**How to reproduce it (as minimally and precisely as possible)**:
I followed the setup with keadm instructions in the documentations
https://kubeedge.io/en/docs/setup/keadm/
**Anything else we need to know?**:
- CentOS 7 in VirtualBox with two network cards working in different modes，the one is NAT,  and the other is Host-Only.
**Environment**:
- Kubernetes version (use `kubectl version`): v1.19.0
- KubeEdge version(e.g. `cloudcore --version` and `edgecore --version`): v1.6.1

**Comment by redplanet00** [comment]
Hi,
I have the exact same problem (Error: CustomResourceDefinition.apiextensions.k8s.io "devices.devices.kubeedge.io" is invalid).
I am using Ubuntu 20.04 with the same Kubernetes and KubeEdge version.
As far as I know KubeEdge worked for me last week, and I haven't changed anything since.

**Comment by fisherxu** [comment]
Have been fixed by #2799 , please try again :)
/close

**Comment by Justin-Young** [comment]
Thanks a lot!
